src/main/java/picocli/CommandLine.java

```diff
@@ -5484,32 +5483,32 @@ public Help create(CommandSpec commandSpec, Help.ColorScheme colorScheme) {
                 Callable<?> callable = Callable.class.cast(cls.getConstructor(Object.class, Object.class).newInstance(null, null));
                 try { return (T) callable.call(); }
                 catch (Exception ex) { throw new InitializationException("Error in Groovy closure: " + ex); }
-
             }
-            if (cls.isInterface() && Collection.class.isAssignableFrom(cls)) {
-                if (List.class.isAssignableFrom(cls)) {
-                    return cls.cast(new ArrayList<Object>());
-                } else if (SortedSet.class.isAssignableFrom(cls)) {
-                    return cls.cast(new TreeSet<Object>());
-                } else if (Set.class.isAssignableFrom(cls)) {
-                    return cls.cast(new LinkedHashSet<Object>());
-                } else if (Queue.class.isAssignableFrom(cls)) {
-                    return cls.cast(new LinkedList<Object>()); // ArrayDeque is only available since 1.6
+            if (cls.isInterface()) {
+                if (Collection.class.isAssignableFrom(cls)) {
+                    if (List.class.isAssignableFrom(cls)) {
+                        return cls.cast(new ArrayList<Object>());
+                    } else if (SortedSet.class.isAssignableFrom(cls)) {
+                        return cls.cast(new TreeSet<Object>());
+                    } else if (Set.class.isAssignableFrom(cls)) {
+                        return cls.cast(new LinkedHashSet<Object>());
+                    } else if (Queue.class.isAssignableFrom(cls)) {
+                        return cls.cast(new LinkedList<Object>()); // ArrayDeque is only available since 1.6
+                    } else {
+                        return cls.cast(new ArrayList<Object>());
+                    }
+                }
+                if (SortedMap.class.isAssignableFrom(cls)) {
+                    return cls.cast(new TreeMap<Object, Object>());
+                }
+                if (Map.class.isAssignableFrom(cls)) {
+                    return cls.cast(new LinkedHashMap<Object, Object>());
                 }
-                return cls.cast(new ArrayList<Object>());
-            }
-            if (Map.class.isAssignableFrom(cls)) {
-                try { // if it is an implementation class, instantiate it
-                    return cls.cast(cls.getDeclaredConstructor().newInstance());
-                } catch (Exception ignored) { }
-                return cls.cast(new LinkedHashMap<Object, Object>());
             }
+            Constructor<T> constructor = cls.getDeclaredConstructor();
             try {
-                @SuppressWarnings("deprecation") // Class.newInstance is deprecated in Java 9
-                T result = cls.newInstance();
```

Comment:
Just wondering why you prefer `Class::newInstance` over `Constructor::newInstance`.

I don't know of any reason to prefer the former over the latter except that it's less of a change, but maybe I'm missing something.

Whereas not using something that's deprecated because it bypasses compile-time exception checking seems like a good reason to prefer the latter over the former.

I'll do whatever you decide, I just want to learn about the pros & cons as I haven't used either of these methods extensively.

Reply:
To be honest I don't remember exactly.
There was a situation where `Constructor.newInstance` gave some error, but `Class.newInstance` worked.
Unfortunately I don't remember the context or environment. 
It feel safer to leave it the way it is.

Reply:
Good enough reason to revert back to `Class` instead of `Constructor`. Will resubmit the PR sometime today with the change.

If you ever remember the situation where the latter errors while the former doesn't, please let me know so I can learn about the differences, too. Thanks.

Reply:
Pushed requested changes